sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -1188,6 +1198,30 @@ def _ITE(*args):
     raises(ValueError, lambda: _ITE((a, x < 2), (b, x > 3)))
 
 
+def test_Piecewise_rewrite_as_sign():
+    from itertools import permutations
+    mult = lambda f, args: [(f*e, c) for e, c in args]
+    strict_args = [((1, x > 0), (0, Eq(x, 0)), (-1, x < 0))]
+    approx_args = [((1, x >= 0), (-1, x < 0)), ((1, x > 0), (-1, x <= 0))]
+    strict_args = [p_arg for arg in strict_args for p_arg in permutations(arg)]
+    approx_args = [p_arg for arg in approx_args for p_arg in permutations(arg)]
+    strict_args += [arg[:-1]+((arg[-1][0], True),) for arg in strict_args]
+    approx_args += [arg[:-1]+((arg[-1][0], True),) for arg in approx_args]
+
+    for args in strict_args:
+        assert Piecewise(*args).rewrite(sign) == sign(x)
+        assert Piecewise(*mult(-1, args)).rewrite(sign) == -sign(x)
+        assert Piecewise(*mult(2, args)).rewrite(sign) == 2*sign(x)
+        assert Piecewise(*mult(x, args)).rewrite(sign) == x*sign(x)
+        assert Piecewise(*mult(-x, args)).rewrite(sign) == -x*sign(x)
```

Comment:
I find this way of writing tests quite hard to read because the test input is far away from the expected output. I think it reads better if you do something more like:
```python
def test_perm(expected, *args):
    for arg in permutations(args):
        assert Piecewise(*arg) == expected

test_perm(sign(x), (1, x > 0), (0, Eq(x, 0)), (-1, x < 0))
test_perm(-sign(x), (1, x >= 0), (-1, x < 0)), ((1, x > 0), (-1, x <= 0))
...
```

Reply:
Okay! It is do.